CLWrapper.cpp

```diff
@@ -71,13 +71,10 @@ void CLWrapper::copyToHost() {
         throw std::runtime_error("copyToHost(): not on device");
     }
 //    cl->finish();
-    cl_event event = NULL;
-    error = clEnqueueReadBuffer(*(cl->queue), devicearray, CL_TRUE, 0, getElementSize() * N, getHostArray(), 0, NULL, &event);    
+    error = clEnqueueReadBuffer(*(cl->queue), devicearray, CL_TRUE, 0, getElementSize() * N, getHostArray(), 0, NULL, NULL);    
     cl->checkError(error);
-    cl_int err = clWaitForEvents(1, &event);
-    clReleaseEvent(event);
-    if (err != CL_SUCCESS) {
-        throw std::runtime_error("wait for event on copytohost failed with " + easycl::toString(err) );
+    if (error != CL_SUCCESS) {
```

Comment:
Following the documentation of OpenCL it should work the way it is now
https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clEnqueueReadBuffer.html